src/pages/sym.mdx

```diff
@@ -0,0 +1,132 @@
+Symbol files, or "sym files" for short, aim to be companions to Game Boy ROM files that provide exactly that: symbol information.
+
+Sym files have been around for a long time (the [no$gmb](http://problemkaputt.de/gmb.htm) emulator supported them, so it is safe to assume since at least the early 2000's), but the format was always loosely defined, leading to subtle incompatibilities.
+This document aims to standardize the format and usage of sym files in order to avoid this problem.
+Modifications, etc. should be suggested by [opening an issue](https://github.com/gbdev/rgbds-www/issues) or [starting a discussion](https://github.com/gbdev/rgbds-www/discussions) on this website's source repository, whichever is more appropriate.
+
+Extensions to this format are known to exist ([example](https://github.com/mattcurrie/mgbdis/#symbol-files)), but they tend to be specific to a certain use case, and are generally a backwards-compatible superset of what are described below.
+Keep in mind that some sym files are written manually, such as when disassembling ROMs, so it is recommended for consumers to be lenient in what they accept.
+
+Note also that [wla-dx's sym files](https://wla-dx.readthedocs.io/en/latest/symbols.html) are much more complex (and complete) than this format.
+RGBDS aims to keep sym files simple, and instead chooses to output other data in different files, such as map files[^1].
+
+[^1]: Map files remain thus far unspecified. Very few tools rely on information contained in them, so no specification has been written in order to keep that format extensible.
+
+## File format specification
+
+:::info
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in [RFC 2119](https://datatracker.ietf.org/doc/html/rfc2119).
+
+Numbers are written in decimal by default, or hexadecimal if prefixed with a `$`, unless otherwise specified.
+
+:::
+
+### Tokenization
+
+Line endings can be either LF-only or CRLF, though the former is preferred; they need not be consistent throughout the document.
+The character encoding must be well-formed UTF-8 (without BOM), however it is strongly encouraged to stick to ASCII characters for portability.
+
+Especially since they are intended to be somewhat human-readable, sym files can contain comments.
+Comments begin with a semicolon (`;`, U+003B) anywhere in a line, all the way to the end of that line.
+Semicolons cannot be escaped, as they are not valid in any part of entries.
+Comments should be ignored by the implementation.
+
+After removing the comment (if any), the line shall be split into non-empty **tokens**, separated by one or more consecutive **whitespace** characters.
+ASCII spaces (` `, U+0020) must be treated as whitespace; implementations may also treat tabs (`\t`, U+0009) as whitespace, but this is not portable to some older implementations.
```

Comment:
Again, traditional implementations don't support spaces, and that's what you should use for portability; however, many humans would end up using tabs because that's practical to them or w/e, and they should be able to if their emulator supports them. This is also why tabs are *always* excluded from symbol names below.

Reply:
That's why I say this is the worst of both words. Just mandate spaces, then! If an implementation wants to extend the format by supporting something invalid, that's fine; that's what we call a format extension. But the specification doesn't need that.

Since a file with tabs in it is invalid by the spec, a strict implementation would error out on it. Let implementations silently handle those files however they wish; don't muddy up the spec. Instead, teach those humans to use spaces instead of tabs, the same way we teach them to not do pointer arithmetic on `void *` in C even though their compiler supports it.

Reply:
The rest of the spec is written to allow tabs as whitespace (e.g. making them part of the character set excluded from tokens), so it wouldn't make sense to allow spaces only. But requiring tabs makes older implementations non-conformant, and I want to avoid friction. Hence this.

Reply:
Making older implementations non-conformant is fine. They can update. The ones that can't will remain non-conformant, and that's fine, because they didn't have time machines and we don't expect a tool from 2004 to conform to a specification written in 2022.
Old, unupdatable implementations will always fall behind the state of the art. That's normal, and it's the users' choice to use such an implementation, so long as they are aware of this difference. Considering that this is going to be used by the same community that offers SDCC/GBDK as a toolchain, a C compiler that is only half conformant to C89 and doesn't even attempt to conform to later standards, I'd say this isn't inherently a problem.

Reply:
beware's gonna hate me, but you have a point. I'll make tabs a requirement.